chaoscenter/web/src/views/ChaosInfrastructureReferenceField/ChaosInfrastructureReferenceField.tsx

```diff
@@ -153,79 +203,163 @@ function ChaosInfrastructureReferenceFieldView({
         }}
         className={cx(css.referenceSelect, css.dialog)}
         title={
-          <Layout.Horizontal spacing={'small'} flex={{ distribution: 'space-between' }}>
+          <Layout.Horizontal>
             <Text font={{ variation: FontVariation.H3 }}>{getString('selectChaosInfrastructure')}</Text>
           </Layout.Horizontal>
         }
       >
-        <Layout.Vertical
-          flex={{ alignItems: 'flex-start', justifyContent: 'flex-start' }}
-          padding={{ bottom: 'medium' }}
-          height={'calc(100% - 32px)'}
-          className={css.gap4}
-        >
-          <Container width={'100%'}>
-            <ExpandingSearchInput
-              alwaysExpanded
-              throttle={500}
-              placeholder={getString('search')}
-              onChange={e => setSearchInfrastructure(e)}
-            />
-          </Container>
-          <Loader
-            loading={loading.listChaosInfra}
-            noData={{
-              when: () => !infrastructureList,
-              messageTitle: getString('noData.title'),
-              message: getString('noData.message')
-            }}
-          >
-            <Layout.Vertical width="100%" className={css.agentList}>
-              <Layout.Vertical width="100%" className={css.agentListInnerContainer}>
-                {infrastructureList && infrastructureList.length > 0 ? (
-                  infrastructureList.map(infrastructure => listItem({ infrastructure: infrastructure }))
-                ) : (
-                  <Layout.Vertical flex={{ justifyContent: 'center' }} spacing="medium" height={'100%'}>
-                    <img src={FallbackBox} alt={getString('latestRun')} />
-                    <Text font={{ variation: FontVariation.BODY1 }} color={Color.GREY_500}>
-                      {searchInfrastructure === ''
-                        ? getString('newUserNoInfra.title')
-                        : getString('noFilteredActiveInfra')}
-                    </Text>
-                    {searchInfrastructure === '' && (
-                      <Button
-                        variation={ButtonVariation.PRIMARY}
-                        text={getString('enableChaosInfraButton')}
-                        onClick={() => {
-                          history.push(paths.toEnvironments());
-                        }}
-                      />
+        {environmentList && environmentList.length > 0 ? (
```

Comment:
```suggestion
        {!!environmentList.length ? (
```

Reply:
getting error in this changes ('environmentList' is possibly 'undefined')